### ISSUE TITLE
Run entire single omic pipeline

### ISSUE THREAD

**Issue by annapamma** [PNNL-CompBio/decomprolute#168]
Focus on perfEval

**Comment by sgosline** [comment]
No the figures are only produced in the three 'metrics' tests that i presented at the CPTAC meeting. Do you want to try those?

**Comment by annapamma** [comment]
Can you give me an example command? I'm a bit confused about how it finds the results from the deconvolution steps. 

Thanks!

**Comment by sgosline** [comment]
So those three commands do not work?

**Comment by annapamma** [comment]
I tried the first one and got the following error:

```
ERROR Tool definition failed initialization:
Tool definition
file:///Volumes/Samsung_T3/tumor_deconvolution/proteomicsTumorDeconv/metrics/data-sim/call-deconv-on-sim.cwl
failed validation:
metrics/data-sim/call-deconv-on-sim.cwl:48:1:   checking field `steps`
metrics/data-sim/call-deconv-on-sim.cwl:85:3:     checking object

`metrics/data-sim/call-deconv-on-sim.cwl#matrix-distance`
metrics/data-sim/call-deconv-on-sim.cwl:86:6:       Field `run` contains
undefined reference to

`file:///Volumes/Samsung_T3/tumor_deconvolution/proteomicsTumorDeconv/metrics/comparison/deconv-comparison-tool.cwl`
```

**Comment by sgosline** [comment]
Good catch, fixing.

**Comment by annapamma** [comment]
I ran this command:

```
cwltool metrics/mrna-prot/mrna-prot-comparison.cwl metrics/mrna-prot/pan-can-matrix.yml
```

And it seemed to hang for a really long time here:

```
INFO [workflow deconv-prot_12] starting step run-deconv_24
INFO [step run-deconv_24] start
INFO [workflow run-deconv_24] start
INFO [workflow run-deconv_24] starting step run-epic_24
INFO [step run-epic_24] start
INFO [job run-epic_6] /private/tmp/docker_tmpj_mbyse0$ docker \
    run \
    -i \
    --mount=type=bind,source=/private/tmp/docker_tmpj_mbyse0,target=/sSVXlI \
    --mount=type=bind,source=/private/tmp/docker_tmpldx7qsa4,target=/tmp \
    --mount=type=bind,source=/private/tmp/docker_tmpxyn0g75s/file.tsv,target=/var/lib/cwl/stg243d76a7-bdc8-43c6-b6ac-e3523af37626/colon-tumor-prot-raw.tsv,readonly \
    --mount=type=bind,source=/Volumes/Samsung_T3/tumor_deconvolution/proteomicsTumorDeconv/signature_matrices/LM7c.txt,target=/var/lib/cwl/stgc09afa6d-518d-49e5-8143-c50f3d9563f2/LM7c.txt,readonly \
    --workdir=/sSVXlI \
    --read-only=true \
    --net=none \
    --user=501:20 \
    --rm \
    --cidfile=/private/tmp/docker_tmp4cimcm8g/20211119103552-133396.cid \
    --env=TMPDIR=/tmp \
    --env=HOME=/sSVXlI \
    tumordeconv/epic \
    Rscript \
    --vanilla \
    /bin/epic.r \
    /var/lib/cwl/stg243d76a7-bdc8-43c6-b6ac-e3523af37626/colon-tumor-prot-raw.tsv \
    /var/lib/cwl/stgc09afa6d-518d-49e5-8143-c50f3d9563f2/LM7c.txt
Warning messages:
1: In EPIC(bulk = df, reference = epicRef) :
  The optimization didn't fully converge for some samples:
01CO005; 01CO013; 01CO014; 01CO022; 05CO037; 05CO041; 05CO048; 05CO054; 09CO006; 09CO008; 09CO013; 11CO005; 11CO007; 11CO010; 11CO020; 11CO021; 11CO036; 11CO042; 11CO044; 11CO047; 11CO051; 11CO057; 11CO058; 11CO061; 11CO062; 14CO003; 16CO011; 20CO003; 20CO007; 21CO006; 22CO004
 - check fit.gof for the convergeCode and convergeMessage
2: In EPIC(bulk = df, reference = epicRef) :
  mRNA_cell value unknown for some cell types: B cells, T4 cells, T8 cells, DC, Granulocytes, MO, NK cells - using the default value of 0.4 for these but this might bias the true cell proportions from all cell types.

```

**Comment by annapamma** [comment]
Actually it's possible that my computer just ran out of memory. Let me try again.

**Comment by sgosline** [comment]
I'll give it a try in a bit. Also, docker images can pollute your machine like crazy. Be sure to run 'docker image prune -a' occassionally.

**Comment by annapamma** [comment]
OK I got a new computer with much higher specs than my last one, and pulled the repo from scratch.  

I tested this command: 
```
cwltool metrics/mrna-prot/mrna-prot-comparison.cwl metrics/mrna-prot/pan-can-matrix.yml
```

and got this error:

```
INFO [job download-prot_3] Max memory used: 14647MiB
ERROR Expecting value: line 1 column 1 (char 0)
script was:
01 "use strict";
02 var inputs = {
03     "cancerType": "luad",
04     "sampleType": "tumor"
05 };
06 var self = [];
07 var runtime = {
08     "cores": 1,
09     "ram": 1024,
10     "tmpdirSize": 1024,
11     "outdirSize": 1024,
12     "tmpdir": "/tmp",
13     "outdir": "/qaIdbl"
14 };
15 (function(){
16   var name = inputs.cancerType + '-' + inputs.sampleType + '-' + 'prot-raw.tsv'
17   self[0].basename = name;
18   return self[0]
19   })()
stdout was: ''
stderr was: 'evalmachine.<anonymous>:17
  self[0].basename = name;
                   ^

TypeError: Cannot set properties of undefined (setting 'basename')
    at evalmachine.<anonymous>:17:20
    at evalmachine.<anonymous>:19:5
    at Script.runInContext (node:vm:139:12)
    at Script.runInNewContext (node:vm:144:17)
    at Object.runInNewContext (node:vm:298:38)
    at Socket.<anonymous> ([eval]:11:57)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:293:11)
    at Socket.Readable.push (node:internal/streams/readable:234:10)'
```

I think it's possible that the docker images being referenced in the main branch aren't updated. I'm pulling them for the first time so ideally they should be the most up-to-date. 

Will look into this.

**Comment by sgosline** [comment]
Yeah, there are some weird issues I find with wonky errors where things fail unexpectedly. I'm running the code now and it's fine (on run '9' and still going).
